litgpt/deploy/serve.py

```diff
@@ -106,6 +121,42 @@ def encode_response(self, output: torch.Tensor) -> Dict[str, Any]:
         return {"output": decoded_output}
 
 
+class StreamLitAPI(BaseLitAPI):
+    def __init__(self,
+                 checkpoint_dir: Path,
+                 precision: Optional[str] = None,
+                 temperature: float = 0.8,
+                 top_k: int = 50,
+                 top_p: float = 1.0,
+                 max_new_tokens: int = 50):
+        super().__init__(checkpoint_dir, precision, temperature, top_k, top_p, max_new_tokens)   
+
+    def setup(self, device: str):
+        super().setup(device)
+
+    def predict(self, inputs: torch.Tensor) -> Any:
+        # Run the model on the input and return the output.
+        prompt_length = inputs.size(0)
+        max_returned_tokens = prompt_length + self.max_new_tokens
+
+        for block in self.model.transformer.h:
+            block.attn.kv_cache.reset_parameters()
+
+        yield from stream_generate(
+                self.model,
+                inputs,
+                max_returned_tokens,
+                temperature=self.temperature,
+                top_k=self.top_k,
+                top_p=self.top_p,
+                stop_tokens=[self.tokenizer.decode(torch.tensor([self.tokenizer.eos_id]))]
```

Comment:
@carmocca  do you know if there's a more efficient way to get the EOS token? We could save it in 

Technically, we could save it as `self.eos_token` attribute here

https://github.com/Lightning-AI/litgpt/blob/cbbe9cda9f0ad535c471460236cfdb0b4f50e2bd/litgpt/tokenizer.py#L40

But the issue would be that it wouldn't work with existing checkpoint files (users would have to redownload)

Reply:
Im confused. Why dou you want this? You need to pass the token ids not strings: https://github.com/Lightning-AI/litgpt/blob/main/litgpt/chat/base.py#L28

Reply:
Hm yeah, good point.  I got confused because when I initially tried passing ints or a list of ints, I got a 

```
buffer_length = max((len(tokens) for tokens in stop_tokens), default=1)
TypeError: object of type 'int' has no len()
```